### PR TITLE
fix decimal stats rescale in presto-orc tuple domain predicate

### PR DESCRIPTION
**Pull request by sopel39**
FYI: @cberner

### CODE REVIEW

**Comment by cberner** [comment]
@dain could you look at this?

File: presto-spi/src/main/java/com/facebook/presto/spi/type/Decimals.java [hunk 1]
```diff
@@ -248,6 +249,16 @@ public static void writeBigDecimal(DecimalType decimalType, BlockBuilder blockBu
         decimalType.writeSlice(blockBuilder, encodeScaledValue(value));
     }
 
+    public static BigDecimal rescale(BigDecimal value, DecimalType type)
+    {
+        value = value.setScale(type.getScale(), ROUND_UNNECESSARY);
```
Comment (dain) [inline presto-spi/src/main/java/com/facebook/presto/spi/type/Decimals.java]: The old code used `ROUND_HALF_UP`.  Is this a change you want to make?

**Comment by dain** [comment]
@cberner I don't know much about this stuff, but it looks correct to me.

File: presto-spi/src/main/java/com/facebook/presto/spi/type/Decimals.java (hunk 1 above)
Comment (sopel39) [inline presto-spi/src/main/java/com/facebook/presto/spi/type/Decimals.java]: Indeed, it was previously `ROUND_HALF_UP`, but it was used after assertion that the scaled was increased, so there was no rounding (`ROUND_HALF_UP` was irrelevant).

I have changed it to `ROUND_UNNECESSARY` and removed scale increase assertion. This makes sure that the precision is not lost.

**Comment by fmeiser** [comment]
Currently we are facing the following exception. Everything works fine after we applied this PR.

```
Caused by: java.lang.IllegalArgumentException: low must be less than or equal to high
        at com.facebook.presto.spi.predicate.Range.<init>(Range.java:50)
        at com.facebook.presto.spi.predicate.Range.range(Range.java:90)
        at com.facebook.presto.orc.TupleDomainOrcPredicate.createDomain(TupleDomainOrcPredicate.java:141)
        at com.facebook.presto.orc.TupleDomainOrcPredicate.getDomain(TupleDomainOrcPredicate.java:113)
        at com.facebook.presto.orc.TupleDomainOrcPredicate.matches(TupleDomainOrcPredicate.java:68)
        at com.facebook.presto.orc.StripeReader.selectRowGroups(StripeReader.java:354)
        at com.facebook.presto.orc.StripeReader.readStripe(StripeReader.java:132)
        at com.facebook.presto.orc.OrcRecordReader.advanceToNextStripe(OrcRecordReader.java:383)
        at com.facebook.presto.orc.OrcRecordReader.advanceToNextRowGroup(OrcRecordReader.java:340)
        at com.facebook.presto.orc.OrcRecordReader.nextBatch(OrcRecordReader.java:299)
        at com.facebook.presto.hive.orc.OrcPageSource.getNextPage(OrcPageSource.java:262)
        ... 9 more
```

**Comment by cberner** [comment]
Merged. Thanks!

**Pull request closed without merging**